bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,156 @@
+    // Save back to localStorage
+    localStorage.setItem(name, existing.toString());
+};
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.announcementsHelper = announcementsHelper();
+        this.fetchAdminAnnouncements();
+        this.newTitle = Oskari.getMsg('admin-announcements', 'addNewForm');
+        this.state = {
+            announcements: [],
+            title: this.newTitle,
+            active: true,
+            activeKey: []
+        };
+    }
+
+    fetchAdminAnnouncements () {
+        this.announcementsHelper.getAdminAnnouncements(function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
+            } else {
+                this.updateState({
+                    announcements: data.data
+                });
+            }
+        }.bind(this));
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsHelper.saveAnnouncement(data, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, data.id);
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
+        this.announcementsHelper.deleteAnnouncement(this.id, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.deleteFailed'));
+            } else {
+                Messaging.success(getMessage('messages.deleteSuccess'));
+                const newList = [...this.state.announcements];
+                newList.splice(index, 1);
+                // Update accordion with announcements and keep all closed
+                this.updateState({
+                    announcements: newList,
+                    activeKey: [],
+                    updated: false
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+    // Cancel creating a new announcement or editing one. Close all panels.
+    cancel (index, id) {
```

Comment:
You could split this to cancelEditing() and cancelAdding() to make it more readable. That way you probably don't need either of the params since adding is always the last? and editing just removes the activeKey from state no matter the what the params are.